how-to/openldap/ldap-saslauthd-kerberos.md

```diff
@@ -9,47 +9,56 @@ All the following configuration will be on `ldap-server.example.com`.
+
 ## Package installation
-Install saslauthd
+Install `saslauthd` on the OpenLDAP server (`ldap-server.example.com` in this document):
 
-```bash
+```text
 sudo apt install sasl2-bin
 ```
-## Saslauthd service principal
-It is assumed you have an admin principal set up in the realm. We will use `ubuntu/admin@EXAMPLE.COM`.
-Next a principal who will be querying Kerberos is needed. The default for saslauthd is "host"/"hostname.domain"@"REALM". For the basic case, it is important that the hostname and domain be what nslookup on your IP number says it is.
 
 ## Check the hostname
 Get the hostname from the server
-```bash
+```text
 hostname -f
 ```
 Which should give you the hostname of:
 ```text
 ldap-server.example.com
 ```
-Also check the hostname and domain using a reverse lookup.
-```bash
-nslookup (your IP)
+Also check the hostname and domain using a reverse lookup with your IP. For example, if the IP address is `10.10.17.91`:
+```text
+nslookup 10.10.17.91
 ```
-Check the name in the reply
+The reply should look like this:
 ```text
-(Your IP).in-addr.arpa      name = ldap-server.example.com. (your IP) will be in reverse here
+91.17.10.10.in-addr.arpa        name = ldap-server.example.com.
 ```
 If the result is the same as your host's canonical name them all is well. If the domain is missing, the [Fully Qualified Domain Name](https://en.wikipedia.org/wiki/Fully_qualified_domain_name) (FQDN) can be entered in the `/etc/hosts` file.
-```bash
+```text
 sudo vi /etc/hosts
 ```
-Add the FQDN before the short hostname.
+Add the FQDN before the short hostname. Using the same IP as in the previous example, we would have:
 ```text
-(your IP) ldap-server.example.com ldap-server
+10.10.17.91 ldap-server.example.com ldap-server
 ```
 ## Create the saslauthd principal
 
-The hostname portion of the principal must be in lowercase. You won't be creating a password for this principal so you will need to ask the kadmin tool to store the key in an external file on the LDAP server. The default location is `/etc/krb5.keytab` so you need to run the kadmin command as sudo.
+The `saslauthd` daemon  needs a kerberos service principal in order to authenticate itself to the kerberos server. Such principals are created with a random password, and the resulting key is stored in `/etc/krb5.keytab`. For more information about kerberos service principals, please consult [How to configure Kerberos service principals](/how-to/kerberos/configure-service-principals/).
+
+The simplest way to create this principal, and extract the key safely, is to run the `kadmin` tool remotely, instead of on the kerberos server. Since the key needs to be written to `/etc/krb5.keytab`, the tool needs to be run with root privileges. Additionally, since creating a new service principal, as well as extracting its key, are privileged operations, we need an `/admin` instance of a principal in order to be allowed these actions. In this example, we will use `ubuntu/admin`:
```

Comment:
```suggestion
The simplest way to create this principal, and extract the key safely, is to run the `kadmin` tool remotely, instead of on the Kerberos server. Since the key needs to be written to `/etc/krb5.keytab`, the tool needs to be run with root privileges. Additionally, since creating a new service principal, as well as extracting its key, are privileged operations, we need an `/admin` instance of a principal in order to be allowed these actions. In this example, we will use `ubuntu/admin`:
```